binders/kafka-binder/spring-cloud-stream-binder-kafka-reactive/src/main/java/org/springframework/cloud/stream/binder/reactorkafka/ReactorKafkaBinder.java

```diff
@@ -86,28 +81,39 @@
 
 	private KafkaExtendedBindingProperties extendedBindingProperties = new KafkaExtendedBindingProperties();
 
+	private ConsumerConfigCustomizer consumerConfigCustomizer;
+
+	private ProducerConfigCustomizer producerConfigCustomizer;
+
 	public ReactorKafkaBinder(KafkaBinderConfigurationProperties configurationProperties,
 			KafkaTopicProvisioner provisioner) {
 
 		super(new String[0], provisioner, null, null);
 		this.configurationProperties = configurationProperties;
 	}
 
+	public void setConsumerConfigCustomizer(ConsumerConfigCustomizer consumerConfigCustomizer) {
```

Comment:
I was curious as to why these new fields are setters and not added to the constructor? I know this technique prevents breaking change and also avoid constructor proliferation. The former concern should not be an issue as this  is in main only.  Mostly curious though.

Reply:
I did it to make it consistent with the message channel binder, but I have no issues moving them to the ctor - @sobychacko - any comment?

Reply:
I am fine with making it consistent with the MC binder for this PR. Maybe in a follow-up change, we can take a comprehensive look at our injection strategy and adopt constructor-based injection. I am fine either way, though, if you want to make the change here.